d2common/d2fileformats/d2ds1/ds1.go

```diff
@@ -483,3 +495,142 @@ func (ds1 *DS1) loadLayerStreams(br *d2datautils.StreamReader, layerStream []d2e
+	}
+
+	if ds1.Version >= v4 {
+		sw.PushInt32(ds1.NumberOfWalls)
+
+		if ds1.Version >= v16 {
+			sw.PushInt32(ds1.NumberOfFloors)
+		}
+	}
+
+	// Step 2 - encode layers
+	for lIdx := range ds1.layerStreamTypes {
+		layerStreamType := ds1.layerStreamTypes[lIdx]
+
+		for y := 0; y < int(ds1.Height); y++ {
+			for x := 0; x < int(ds1.Width); x++ {
+				dw := uint32(0)
+
+				switch layerStreamType {
+				case d2enum.LayerStreamWall1, d2enum.LayerStreamWall2, d2enum.LayerStreamWall3, d2enum.LayerStreamWall4:
+					wallIndex := int(layerStreamType) - int(d2enum.LayerStreamWall1)
+					dw |= uint32(ds1.Tiles[y][x].Walls[wallIndex].Prop1) & 0xFF            //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].Sequence) & 0x3F) << 8  //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].Unknown1) & 0xFC) << 14 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].Style) & 0x3F) << 20    //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].Unknown2) & 0x7C) << 26 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].hidden) & 0x01) << 31   //nolint:gomnd // Bitmask
+				case d2enum.LayerStreamOrientation1, d2enum.LayerStreamOrientation2,
+					d2enum.LayerStreamOrientation3, d2enum.LayerStreamOrientation4:
+					wallIndex := int(layerStreamType) - int(d2enum.LayerStreamOrientation1)
+					dw |= uint32(ds1.Tiles[y][x].Walls[wallIndex].Type)
+					dw |= (uint32(ds1.Tiles[y][x].Walls[wallIndex].Zero) & 0xFFFFFF00) << 8 //nolint:gomnd // Bitmask
+
+				case d2enum.LayerStreamFloor1, d2enum.LayerStreamFloor2:
+					floorIndex := int(layerStreamType) - int(d2enum.LayerStreamFloor1)
+					dw |= uint32(ds1.Tiles[y][x].Floors[floorIndex].Prop1) & 0xFF            //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Floors[floorIndex].Sequence) & 0x3F) << 8  //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Floors[floorIndex].Unknown1) & 0xFC) << 14 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Floors[floorIndex].Style) & 0x3F) << 20    //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Floors[floorIndex].Unknown2) & 0x7C) << 26 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Floors[floorIndex].hidden) & 0x01) << 31   //nolint:gomnd // Bitmask
+				case d2enum.LayerStreamShadow:
+					dw |= uint32(ds1.Tiles[y][x].Shadows[0].Prop1) & 0xFF            //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Shadows[0].Sequence) & 0x3F) << 8  //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Shadows[0].Unknown1) & 0xFC) << 14 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Shadows[0].Style) & 0x3F) << 20    //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Shadows[0].Unknown2) & 0x7C) << 26 //nolint:gomnd // Bitmask
+					dw |= (uint32(ds1.Tiles[y][x].Shadows[0].hidden) & 0x01) << 31   //nolint:gomnd // Bitmask
```

Comment:
I was confused, because it didn't return correct data, but I've tested it and it doesn't affect on the game.

Reply:
solved by 5702d96